tests/_fixture/CheckoutPage.ts

```diff
@@ -0,0 +1,115 @@
+import { expect, Page } from '@playwright/test';
+import { Polly } from '@pollyjs/core';
+
+export default class CheckoutPage {
+    readonly page: Page;
+    readonly isReplay: boolean;
+    polly: Polly | undefined;
+
+    constructor(page: Page) {
+        this.page = page;
+        this.isReplay = process.env.MODE?.toLowerCase() === 'replay';
+    }
+
+    removeSensitiveHeaders(): void {
+        // hack for BC dev store's root certificate issue during recording HAR
+        // https://stackoverflow.com/questions/31673587/error-unable-to-verify-the-first-certificate-in-nodejs
+        process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0';
+
+        this.polly?.server.any().on('beforePersist', (_, recording) => {
+            const processHeaders = (headers: []): {} => {
+                const dummyData = '*';
+                const sensitiveHeaderNames = [
+                    'authorization',
+                    'cookie',
+                    'set-cookie',
+                    'token',
+                    'x-session-hash',
+                    'x-xsrf-token',
+                ];
+
+                return headers.map((header: { name: string; value: string }) => {
+                    if (sensitiveHeaderNames.includes(header.name)) {
+                        return { ...header, value: dummyData };
+                    }
+
+                    return header;
+                });
+            };
+
+            recording.request.headers = processHeaders(recording.request.headers);
+            recording.response.headers = processHeaders(recording.response.headers);
+        });
+    }
+
+    removeBootstrapRequests(): void {
+        this.polly?.server.any().intercept((req, _, interceptor) => {
+            const ignoredRequests = [
+                'amazonaws.com',
+                'icon-sprite.svg',
+                '/api/storefront/carts/',
+                '/api/storefront/checkout/',
+                '/api/storefront/checkouts/',
+                '/api/storefront/checkout-settings',
+                '/api/storefront/form-fields',
+                '/api/storefront/orders/',
+                '/internalapi/v1/shipping/countries',
+            ];
+            if (ignoredRequests.some(item => req.url.includes(item))) {
+                // console.log('🚫PASS   ' + req.url);
+                interceptor.passthrough();
+            } else {
+                // console.log('😃RECORD ' + req.url);
+                interceptor.abort();
+            }
+        });
+    }
+
+    async createCartOn(storeURL: string): Promise<void> {
```

Comment:
I think we likely need to break this into multiple methods. For example, we could have one method that creates a cart by adding some products. This method could have several variants, i.e.: physical vs digital product, single vs multiple. And then we could have another method to complete the customer step, as a guest or returning customer etc... The reason is because we might need to set up the cart in various combinations. Having smaller, separate methods would allow us to do that instead of having to write one for each combination.

Reply:
Created `ApiRequestsSender` to create a cart via multiple API calls (wip) and checkout environment presets inside `PlaywrightHelper`  to offer a predefined checkout page state.